src/cmap/conn/mod.rs

```diff
@@ -246,11 +246,26 @@ impl Connection {
         self.error = write_result.is_err();
         write_result?;
 
+        self.read_response().await
+    }
+
+    /// Reads a response from the server and returns a `CommandResponse`.
+    ///
+    /// An `Ok(...)` result simply means the server received the command and that the driver
+    /// driver received the response; it does not imply anything about the success of the command
+    /// itself.
+    pub(crate) async fn read_response(&mut self) -> Result<CommandResponse> {
         let response_message_result = Message::read_from(&mut self.stream).await;
         self.command_executing = false;
         self.error = response_message_result.is_err();
 
-        CommandResponse::new(self.address.clone(), response_message_result?)
+        let response_message = response_message_result?;
+
+        if !response_message.more_to_come() {
+            self.command_executing = false;
```

Comment:
Is this assignment necessary? It looks like `command_executing` is already set to false in all cases on line 259 above.

Reply:
Oops, good catch! That first one should be setting it to `true`.

Reply:
Ah wait, no, it needs to just delay setting false until the if statement. Since this method is called by `send_command`, we want to only mark the connection as not being in use once we're sure that there aren't any more messages in the buffer. Still a bug as written though, so good catch!